### ISSUE TITLE
Too many lines of code

### ISSUE THREAD

**Issue by GolanTrevize10** [I8C/sonar-flow-plugin#20]
Hello, we have some problem with some of our flow projects in our sonarQube server. The problem is that the project has 9 millions of lines of code. That is causing us some issues because SonarQube needs a lot of memory to process so many lines. I found out that SonarQube is counting as lines of code also comments and blank lines in the flow files. Is there any way of avoiding that?

**Comment by dewanst** [comment]
Hi
Normally the latest version should only count real lines of code (eg flow). Which version are you using?
Next to that the plugin has to scan/process the complete files to understand its structure and apply rules, so I guess there isn't a way in avoiding that.
What you could do, is to split up your project if possible and scan in chunks.

**Comment by GolanTrevize10** [comment]
Hi dewanst, thank you very much for your answer. We are using the version
0.4 of the plugin. I thought in splitting the project but another problem
we have is that we have sonarqube enterprise license. This license
restricts the total number of lines of code we can have in all our
projects. So a project that counts 9 millions of line of code, 10 times
more than any other project we have, exceeds our license limit. That's why
I was wondering if it is possible to reduce the number of lines counted by
the plugin

**Comment by dewanst** [comment]
Hi GolanTrevize10
I would recommend you use the latest release of our plugin as we did some fixes in this area, and the lines of code should now be correctly counted.
Kr

**Comment by dewanst** [comment]
Hi GolanTrevize10

Any updates on your issue?

**Comment by GolanTrevize10** [comment]
Hi, we analyzed the project again with new plugin version 1.0. But now it is not recognizing any of the flow.xml files as flow files. What is the difference between the Flow file pattern setting and the Flow file suffixes setting. All our flow files are called flow.xml. We don't have any ndf file

**Comment by xxxvodnikxxx** [comment]
about the patterns its #22  related

**Comment by AgentW4C** [comment]
Exactly, you need to set sonar.flow.file.suffixes to 'xml' and remove 'xml' from sonar.xml.file.suffixes in order the files are treated as Flow files and not XML files.
